test/extensions/filters/http/cors/cors_filter_test.cc

```diff
@@ -60,17 +62,17 @@ class CorsFilterTest : public testing::Test {
           return {cors_policy_.get(), cors_policy_.get()};
         }));
 
-    filter_.setDecoderFilterCallbacks(decoder_callbacks_);
-    filter_.setEncoderFilterCallbacks(encoder_callbacks_);
+    filter_->setDecoderFilterCallbacks(decoder_callbacks_);
+    filter_->setEncoderFilterCallbacks(encoder_callbacks_);
   }
 
-  bool isCorsRequest() { return filter_.is_cors_request_; }
+  bool isCorsRequest() { return filter_->is_cors_request_; }
 
   NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks_;
   NiceMock<Http::MockStreamEncoderFilterCallbacks> encoder_callbacks_;
   Stats::TestUtil::TestStore stats_;
   CorsFilterConfigSharedPtr config_;
-  CorsFilter filter_;
+  std::shared_ptr<CorsFilter> filter_;
```

Comment:
Why is this change needed?
If possible, I suggest leaving this as `CorsFilter` type.

Reply:
compiler complain that because now it inherit from passthroughfilter. Compiler will throw an error about multiple assignments of virtual base.

Reply:
here is the error https://github.com/envoyproxy/envoy/actions/runs/11956111680/job/33330127779

I actually dislike to inherit from multiple classes and virtual inheritance. But anyway, it's widely used by envoy and almost all stream filters inherit from passthroughfilter. So, I think it's fine for now, before we can get better abstraction/solution.

Reply:
I guess that the main issue is that the places where the CorsFilter is moved in the tests actually don't need to have the filter as a data member (just create a locally scoped filter `CorsFilter filter(config_);` instead of updating the `filter_`).
As this is just a test code, I don't fell strongly about using a pointer, I just think it makes more sense in this case to not use one.

Reply:
I also didn't feel strongly about that and only did a global replacement.